CMakeLists.txt

```diff
@@ -140,17 +140,32 @@ foreach(HEADER_SRCPATH IN ITEMS ${AWS_COMMON_HEADERS} ${AWS_COMMON_OS_HEADERS} $
     install(FILES ${HEADER_SRCPATH} DESTINATION ${HEADER_DSTDIR})
 endforeach()
 
-install(
-    TARGETS ${CMAKE_PROJECT_NAME} EXPORT ${CMAKE_PROJECT_NAME}-config
-        ARCHIVE DESTINATION ${CMAKE_INSTALL_PREFIX}/lib
-        LIBRARY DESTINATION ${CMAKE_INSTALL_PREFIX}/lib
-        COMPONENT library
-)
+set(LIBRARY_DIRECTORY ${CMAKE_INSTALL_PREFIX}/lib)
 
-export(TARGETS ${CMAKE_PROJECT_NAME} FILE ${CMAKE_PROJECT_NAME}-config.cmake)
-install(EXPORT ${CMAKE_PROJECT_NAME}-config DESTINATION "lib/${CMAKE_PROJECT_NAME}/cmake/")
+install(TARGETS ${CMAKE_PROJECT_NAME}
+    EXPORT ${CMAKE_PROJECT_NAME}-targets
+    ARCHIVE DESTINATION ${LIBRARY_DIRECTORY}
+    LIBRARY DESTINATION ${LIBRARY_DIRECTORY})
+
+configure_file("cmake/${CMAKE_PROJECT_NAME}-config.cmake"
+  "${CMAKE_CURRENT_BINARY_DIR}/${CMAKE_PROJECT_NAME}-config.cmake"
+  @ONLY)
+
+install(EXPORT "${CMAKE_PROJECT_NAME}-targets"
+    DESTINATION "${LIBRARY_DIRECTORY}/${CMAKE_PROJECT_NAME}/cmake/"
+    NAMESPACE AWS::)
+
+install(FILES "${CMAKE_CURRENT_BINARY_DIR}/${CMAKE_PROJECT_NAME}-config.cmake"
+    DESTINATION "${LIBRARY_DIRECTORY}/${CMAKE_PROJECT_NAME}/cmake/")
+
+list(APPEND EXPORT_MODULES
```

Comment:
How much of this needs to be installed? Consumers of this library shouldn't need to use the SIMD-flag-setting logic, for example.

Reply:
This part is from the original script.
@ColdenCullen can speak for it I guess.
My understanding is that those are helpful CMake scripts that can be utilized by other projects, and since aws-c-common is the lowest in the dependency tree, it made sense to have it be part of its installation.